lib/utils.js

```diff
@@ -26,31 +24,37 @@ function populateCache (url) {
       return Winston.error('Non-200 status code: %d', res.statusCode)
     }
 
-    const title = res.body.title
-    const headers = res.body.headers
-
-    const batch = Cache.batch()
-
-    // Add each facility to the cache
-    res.body.rows.forEach(function (row) {
-      const key = row[0]
-      if (!key) {
-        return Winston.error('Facility missing value', {facility: row})
-      }
-      const body = Mustache.render(template, {
-        title: title,
-        headers: JSON.stringify(headers),
-        row: JSON.stringify(row),
-        width: headers.length
+    const levelKey = url.split('sqlViews/')[1] // levelKey = <UID>.json
+    Cache.deleteSubLevelCacheDB(levelKey, () => { // clear the existing cache before add new data
```

Comment:
The way to overcome it is to do the delete as part of the batch that we execute below. It can just be the first operation in the batch and then adding the cache entries will come after that. LevelDB executes the batch atomically so there will be no gap between the delete and adding the new entries where empty results could be returned.